amd_openvx/openvx/ago/ago_haf_cpu_generic_functions.cpp

```diff
@@ -0,0 +1,878 @@
+{
+    vx_status status = VX_SUCCESS;
+    vx_df_image format, filling_format;
+
+    format = VX_DF_IMAGE_U8;
+    vx_image tmp = vxCreateImage(context, width, height, VX_DF_IMAGE_U8);
+    status |= vxQueryImage(filling, VX_IMAGE_FORMAT, &filling_format, sizeof(filling_format));
+
+    vx_rectangle_t tmp_rect, filling_rect;
+    vx_imagepatch_addressing_t tmp_addr = VX_IMAGEPATCH_ADDR_INIT;
+    vx_imagepatch_addressing_t filling_addr = VX_IMAGEPATCH_ADDR_INIT;
+    vx_map_id tmp_map_id, filling_map_id;
+    void *tmp_base = NULL;
+    void *filling_base = NULL;
+
+    status = vxGetValidRegionImage(tmp, &tmp_rect);
+    status |= vxMapImagePatch(tmp, &tmp_rect, 0, &tmp_map_id, &tmp_addr, (void **)&tmp_base, VX_READ_AND_WRITE, VX_MEMORY_TYPE_HOST, 0);
+    status = vxGetValidRegionImage(filling, &filling_rect);
+    status |= vxMapImagePatch(filling, &filling_rect, 0, &filling_map_id, &filling_addr, (void **)&filling_base, VX_READ_AND_WRITE, VX_MEMORY_TYPE_HOST, 0);
+
+    for (int ix = 0; ix < (int)width; ix++)
+    {
+        for (int iy = 0; iy < (int)height; iy++)
+        {
+
+            void* tmp_datap = vxFormatImagePatchAddress2d(tmp_base, ix, iy, &tmp_addr);
+
+            if (iy % 2 != 0 || ix % 2 != 0)
+            {
+                if (format == VX_DF_IMAGE_U8)
+                    *(vx_uint8 *)tmp_datap = (vx_uint8)0;
+                else
+                    *(vx_int16 *)tmp_datap = (vx_int16)0;
+            }
+            else
+            {
+                void* filling_tmp = vxFormatImagePatchAddress2d(filling_base, ix / 2, iy / 2, &filling_addr);
+                vx_int32 filling_data = filling_format == VX_DF_IMAGE_U8 ? *(vx_uint8 *)filling_tmp : *(vx_int16 *)filling_tmp;
+                if (format == VX_DF_IMAGE_U8)
+                {
+                    if (filling_data > UINT8_MAX)
+                        filling_data = UINT8_MAX;
+                    else if (filling_data < 0)
+                        filling_data = 0;
+                    *(vx_uint8 *)tmp_datap = (vx_uint8)filling_data;
+                }
+                else
+                {
+                    if (filling_data > INT16_MAX)
+                        filling_data = INT16_MAX;
+                    else if (filling_data < INT16_MIN)
+                        filling_data = INT16_MIN;
+                    *(vx_int16 *)tmp_datap = (vx_int16)filling_data;
+                }
+            }
+        }
+    }
+
+    status |= vxUnmapImagePatch(tmp, tmp_map_id);
+    status |= vxUnmapImagePatch(filling, filling_map_id);
+
+    status |=replicateConvolve(tmp, conv, upsample, border);
+
+    vx_rectangle_t upsample_rect;
+    vx_imagepatch_addressing_t upsample_addr = VX_IMAGEPATCH_ADDR_INIT;
+    vx_map_id upsample_map_id;
+    void * upsample_base = NULL;
+    vx_df_image upsample_format;
+
+    status |= vxQueryImage(upsample, VX_IMAGE_FORMAT, &upsample_format, sizeof(upsample_format));
+    status = vxGetValidRegionImage(upsample, &upsample_rect);
```

Comment:
Codacy found an issue: [Variable 'status' is reassigned a value before the old one has been used.](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=6287409)